channel/channel.go

```diff
@@ -0,0 +1,57 @@
+package channel
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+)
+
+type CryChan[T any] struct {
+	c chan T
+	t time.Duration
+}
+
+// NewCryChan creates a new cryChan of the given type and size
+func NewCryChan[T any](bufSize int, timeout time.Duration) CryChan[T] {
+	Logger = Logger.With().Int("size", bufSize).Logger()
+
+	return CryChan[T]{
+		c: make(chan T, bufSize),
+		t: timeout,
+	}
+}
+
+// Push adds an element in the channel, or logs if it's blocked for too long
+func (c *CryChan[T]) Push(e T) {
+	start := time.Now()
+	select {
+	case c.c <- e:
+	case <-time.After(c.t):
+		// prints the first 16 bytes of the trace, which should contain at least
+		// the goroutine id.
+		trace := make([]byte, 16)
+		runtime.Stack(trace, false)
+		Logger.Warn().Str("obj", fmt.Sprintf("%+v", e)).
```

Comment:
@nkcr and @pierluca : shouldn't we use debug.Stack() instead ?

Reply:
Why would you do that ? :) 
we are only interested in having the goroutine id here.

Reply:
I think I agree with @jbsv, unless I'm missing something.

How are you going to debug where the timeout occurred with just the goroutine id ?